src/arrayMethodJoin.js

```diff
@@ -5,7 +5,24 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let res = '';
+
+    for (let i = 0; i < this.length - 1; i++) {
+      if (separator === undefined) {
+        res += `${this[i]},`;
+      } else if (this[i] === undefined || this[i] === null) {
+        this[i] = separator;
+        res += this[i];
+      } else {
+        res += `${this[i]}${separator}`;
+      }
+    }
+
+    if (this.length === 0) {
+      return '';
```

Comment:
```suggestion
      return res;
```
в тебе там буде пуста строка і так